### PR TITLE
Added task to cleanup Edge regkeys and folders

### PR DESCRIPTION
**Pull request by mustjab**
This change fixes issue tracked in #19878 The fix is to cleanup Edge registry and files that can get left behind if uninstall fails. As part of this change we also copy setup log file to help analyze failure and hopefully prevent it from happening in the future.

The change is in cleanup code, requires two runs to confirm it:

Run1 (expect failures): https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=35216
Run2 (expect it to pass): https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=35218

Dev channel run: https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=35226

### CODE REVIEW

File: tools/ci/azure/cleanup_win10.yml [hunk 1]
```diff
@@ -50,7 +57,41 @@ steps:
         Restart-Computer -Force
       }
     }
+    if (($stableExitCode -ne $exitCodeSuccess) -and
+        ($canaryExitCode -ne $exitCodeSuccess) -and
```
Comment (foolip) [inline tools/ci/azure/cleanup_win10.yml]: Nit: Order stable, beta, dev, canary would be nice, or perhaps alphabetical. Same above and below.

File: tools/ci/azure/cleanup_win10.yml [hunk 2]
```diff
@@ -50,7 +57,41 @@ steps:
...
+      Remove-Item -Path "$env:localappdata\Microsoft\EdgeUpdate" -Recurse -Force -ErrorAction SilentlyContinue
+      $edgePath = "$env:localappdata\Microsoft\$folder"
+      Remove-Item -Path $edgePath -Recurse -Force -ErrorAction SilentlyContinue
+      $edgePath = "$env:systemdrive\Program Files (x86)\Microsoft\$folder"
+      Remove-Item -Path $edgePath -Recurse -Force -ErrorAction SilentlyContinue
+    }
+  displayName: 'Cleanup Edge registry'
+  condition: always()
+  errorActionPreference: silentlyContinue
+  ignoreLASTEXITCODE: true
```
Comment (foolip) [inline tools/ci/azure/cleanup_win10.yml]: Nit: missing final newline

File: tools/ci/azure/cleanup_win10.yml [hunk 3]
```diff
@@ -50,7 +57,41 @@ steps:
...
+    } else {
+      Write-Host "Successfully uninstalled all Edge channels"
+    }
   displayName: 'Restore hosts file and cleanup test machine'
   condition: always()
   errorActionPreference: silentlyContinue
   ignoreLASTEXITCODE: true
+
+- powershell: |
+    $channelFolders = @('Edge', 'Edge SxS', 'Edge Dev', 'Edge Beta')
```
Comment (foolip) [inline tools/ci/azure/cleanup_win10.yml]: Maybe just `$channels`? These aren't actually folder names as clear inside the loop, but the suffix.

**Review by foolip — APPROVED** [review]
A few nits, but looks good, feel free to merge when trial runs show promise. Is it necessary to also trial run Edge Dev or are you confident this can't break them?

File: tools/ci/azure/cleanup_win10.yml (hunk 2 above)
Comment (mustjab) [reply tools/ci/azure/cleanup_win10.yml]: Fixed

File: tools/ci/azure/cleanup_win10.yml (hunk 3 above)
Comment (mustjab) [reply tools/ci/azure/cleanup_win10.yml]: Fixed

File: tools/ci/azure/cleanup_win10.yml (hunk 1 above)
Comment (mustjab) [reply tools/ci/azure/cleanup_win10.yml]: Fixed

File: tools/ci/azure/cleanup_win10.yml [hunk 4]
```diff
@@ -72,26 +72,26 @@ steps:
   ignoreLASTEXITCODE: true
 
 - powershell: |
-    $channelFolders = @('Edge', 'Edge SxS', 'Edge Dev', 'Edge Beta')
-    foreach ($folder in $channelFolders) {
-      Write-Host "Deleting regkeys for $folder folder"
-      Remove-Item -Path "HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\Microsoft $folder" -Recurse -Force -ErrorAction SilentlyContinue
-      Remove-Item -Path "HKCU:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Microsoft $folder" -Recurse -Force -ErrorAction SilentlyContinue
+    $channels = @('Edge', 'Edge Beta', 'Edge Dev', 'Edge Canary')
```
Comment (foolip) [inline tools/ci/azure/cleanup_win10.yml]: Canary was SxS before, was that an intentional change?

Comment (mustjab) [reply tools/ci/azure/cleanup_win10.yml]: Not intentional, good catch, thank you!

**Review by foolip — APPROVED** [review]

**Pull request merged**